### PR TITLE
Introduce binary compatibility validator

### LINKED ISSUE #168: Introduce Binary compatibility validator to track API changes
https://github.com/Kotlin/binary-compatibility-validator

### PR DESCRIPTION
**Pull request by Goooler**
Closes #168.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -21,6 +22,16 @@ infra {
     }
 }
 
+apiValidation {
+    ignoredPackages += listOf(
+        "kotlinx.collections.immutable.internal"
```
Comment (qurbonzoda) [inline build.gradle.kts:27]: The package doesn't seem to have any public declaration currently. But I think it is important to catch unintended public declarations in this package in the future.

**Review by qurbonzoda — CHANGES_REQUESTED** [review]
I would suggest validating public declarations in the `kotlinx.collections.immutable.internal` package as well. Otherwise, LGTM

File: build.gradle.kts (hunk 1 above)
Comment (Goooler) [reply build.gradle.kts]: Done.

**Comment by qurbonzoda** [comment]
Thanks for the contribution!

**Pull request merged**